src/ext/java/org/opentripplanner/ext/siri/AffectsMapper.java

```diff
@@ -91,117 +94,105 @@ private List<EntitySelector> mapVehicleJourneys(AffectsScopeStructure.VehicleJou
         }
       }
 
+      LocalDate serviceDate = null;
+
+      if (affectedVehicleJourney.getOriginAimedDepartureTime() != null) {
```

Comment:
This should probably be resolved from either `FramedVehicleJourneyRef` or `DatedVehicleJourneyRef` if defined. Good candidate to push to `EntityResolver`.

Reply:
The usage of this field is part of the "temporary solution" to support SIRI SX-messages tagged on  trainNumber/privateCode + date. The trainNumber is passed in `VehicleJourneyRef`, and the departuredate in `OriginAimedDepartureTime`.

When `FramedVehicleJourneyRef` is provided, the ServiceDate is fetched from there instead - using `EntityResolver` ([line 149](https://github.com/opentripplanner/OpenTripPlanner/pull/4888/files/08544150545f1dd9d7b0a415ed61fe2afe0b56e6#diff-c2b0521a62568e59a92789a535de146ede2523ed49fdf9a5a444eee84e0d2be2R149)).

When `DatedVehicleJourneyRef` is provided, the `OriginAimedDepartureTime` _should not_ be used, but it will be if provided.

The reason for supporting all the different types/combinations is that it is not OTP's responsibility to enforce that the types of data that is referenced is of the correct type as this may differ depending on which SIRI-profile is used in different installations.

Reply:
> The usage of this field is part of the "temporary solution" to support SIRI SX-messages tagged on trainNumber/privateCode + date. The trainNumber is passed in `VehicleJourneyRef`, and the departuredate in `OriginAimedDepartureTime`.

Could we move the code closer to the usage in that case, even all the way to the fuzzy matcher?